source/extensions/filters/udp/udp_proxy/udp_proxy_filter.h

```diff
@@ -294,7 +301,14 @@ class HttpUpstreamImpl : public HttpUpstream, protected Http::StreamCallbacks {
     }
 
     void decodeHeaders(Http::ResponseHeaderMapPtr&& headers, bool end_stream) override {
-      bool is_valid_response = Http::CodeUtility::is2xx(Http::Utility::getResponseStatus(*headers));
+      auto status = Http::Utility::getResponseStatus(*headers);
+      bool is_valid_post = parent_.tunnel_config_.usePost() && Http::CodeUtility::is2xx(status);
+      // In connect-udp case, Envoy will transform the H2 headers to H1 upgrade headers.
+      // A valid response should have SwitchingProtocol status and connect-udp upgrade.
+      bool is_valid_upgrade = status == 101 && Http::HeaderUtility::isConnectUdp(*headers);
```

Comment:
Can we move the `status == 101` check to inside `Http::HeaderUtility::isConnectUdp()`?